### DISCUSSION TITLE
New Meshtastic gadgets

### DISCUSSION THREAD

**Discussion by kevin-seeed** [Seeed-Studio/OSHW-XIAO-Series#16] [Open for Vote]
<img width="960" alt="image" src="https://github.com/Seeed-Studio/XIAO_Series/assets/170942971/ef052a05-db32-44e8-872e-9c690bec788c">

**Comment by mdroberts1243** [comment]
One improvement is like to see is in solar charging circuit. In the winter the charging circuit shouldn't attempt to charge the battery if below 0 degrees C.

**Reply by kevin-seeed** [reply]
correct, and low temperature also kills lipo batteries.

**Reply by Pixeldieb** [reply]
Alterntively monitor temp and heat up the case. This is done in the SG-50 by Milesight and i love that it enables me to run it in german winters with no issues.

**Comment by chalei** [comment]
The board is already good on itself for meshtastic. but If I may suggest, the screen is better to use an oled or epaper display for low power consumption

**Reply by meilily** [reply]
EPaper sounds great～

**Reply by kkwestt** [reply]
add EPaper version )

**Reply by CWGwaltney** [reply]
https://www.seeedstudio.com/ePaper-Breakout-Board-p-5804.html

**Comment by CWGwaltney** [comment]
Yes... also include IIC Battery Management Chip that can report and control Power

**Reply by meilily** [reply]
Roger!

**Comment by kkwestt** [comment]
I don't think there are enough **pins** to connect the display

**Reply by kevin-seeed** [reply]
that's why we need an IO expansion chip.

**Reply by CWGwaltney** [reply]
https://www.seeedstudio.com/ePaper-Breakout-Board-p-5804.html

**Reply by CWGwaltney** [reply]
I wonder if this device can drive epaper or regular screen via same ribbon cable

**Comment by CWGwaltney** [comment]
Use this form factor

https://www.seeedstudio.com/Wio-Tracker-1110-Dev-Board-p-5799.html

**Comment by Pixeldieb** [comment]
Feedback:

### DISPLAY AND CASE
- easier display that is also able to be extended by cable (for mounting in different cases)
That would also add to the repairability if its not a special design display but a common one.
I personally would prefer an e-paper display as the power consumption is way lower and it could be used to identify the device even when power isnt used. Best would be to display basic data of the node. 
That said i would love to see an option with detachable screen to be able to integrate the node in for example the m2 multiplattform gateway with very little modifications.

- buttons on the same side with laser cut out files (stl) for manufacturing device enclosures to fit the buttons

#### Connection options

- Onboard bluetooth and gps antenna to lessen the steps needed to build with the kit
- connectors for LoRa and Bluetooth at the edge of board or with a pre-planned cable holder / cable channel / cable clips to neatly mount the extension cables
- 2ph 2mm color coded connector for battery / OR screw terminal to accept cables directly (adds depth, makes it easier to install on wide variety)
- 2ph 2mm color coded connector for solar panel / OR screw terminal to accept cables directly (adds depth, makes it easier to install on wide variety)
- USB Option for emergency powering and connecting for troubleshooting (Data and Power)

#### LEDs
I would wish for at least 3 LEDs for showcasing:
1. Working Mode (running, fault, off)
2. Solar Charging mode (on, off)
3. Color coded LED for battery % or self programmable options

#### Mounting
V-Lock Style / dovetail attachment on the back of case to accomodate a wider range of mounting options. Especially for solar it would be preferred to use angled brackets. If the back of the design uses a gravity fed and screw secured V-Lock it is easy to design different mounting options for the case. If you use a 1/4 thread for the secure screw hole then also common camera gear can be used to mount the node.

###  SOFTWARE
- pre loaded meshtastic firmware configured as Client without MQTT enabled

### ENERGY AND POWER
- solar charge controller as mentioned by so many before.
This is important, as the device should be able to work with different lithium cells. Most common are the 18650 in paralell.
Sample design of current solar nodes
<img width="555" alt="Bildschirmfoto 2024-11-07 um 12 17 01" src="https://github.com/user-attachments/assets/f4ac797d-e92a-46af-8a41-8d8c7a74ae4a">
<img width="555" alt="Bildschirmfoto 2024-11-07 um 11 56 20" src="https://github.com/user-attachments/assets/9be1516b-5a4c-46aa-91f3-42b53ead851f">
On the left there is the communication part. In the middle a paralell block of 18650 cells (should be replacable to accomodate different power options). On the right the solar charge controller.


- step up / booster for the power from the solar or batteries

- automatic shutoff to save power, especially since we are solar powered here
This is needed to because for german winters there is most often not enough sun to power.

- charge protection
especially lithium cells shouldnt be just charged until full to expand battery life. A simple lithium charger would prevent that.

- Temperature protection
An onboard temperature sensor with a small heating element to prevent device from freezing when in low energy mode in cold climates.

- solar charging should be smoothened out and be primarily used before taking energy from the battery
- USB-C Charging as an option to charge the battery in low power situations OR multiple 18650 cells in a parallel holder to exchange batteries when device doesnt get enough sun in winter.

- on/off switch as an option (ideally to solder a 2pin button on or bridge the gap with solder for constant power supply)

### Bonus
Being able to use a cheap all-in-one keyboard with trackpad to connect via bluetooth to control the device and send actual messages would be dope. That would need a designated bluetooth pairing button, a bluetooth LED for status and some extra firmware.

<img width="335" alt="Bildschirmfoto 2024-11-07 um 16 32 59" src="https://github.com/user-attachments/assets/d40e2544-8ae4-4701-9c04-b52a8dd92c9c">
